javascript/entities/player.js

```diff
@@ -3,6 +3,7 @@
 // be found in the LICENSE file.
 
 import PlayerSyncedData from 'entities/player_synced_data.js';
+import PlayerSettings from 'features/player_settings/player_settings.js';
```

Comment:
Can this live in `entities/player_settings.js`?

We have some layering in our code to keep a few concepts separated. In a nutshell:

- **base/** can only depend on other things in `base/`.
- **entities/** can depend on things in `base/` and other things in `entities/`.
- **components/** can depend on things in `base/`, `entities/` and `components/`.
- **features/** can depend on `base/`, `entities/` and `components/`, as well as other code in `features/` as long as the class that extends `Feature` calls `this.defineDependency()` to get the instance. We do this to avoid circular dependencies.

The easiest way to change PlayerSetting to allow for that probably is to _not_ inherit from `Setting` leading to a bit of duplication, or to move `Setting` to `base/`. I'm fine with either.

I'm less certain about `PlayerSettings`. My suggestion would be to have `PlayerSettings` be a pure and simple list, which would be fine for `base/`.

The `PlayerSettingsCommands` could then do both _(a)_ the database operations, and own the `PlayerSettingsDatabase` instance, and _(b)_ updating the values in the player's setting list.

Reply:
The `PlayerSetting` should still inherit from `Setting`. I will move `Setting` to entities too.
I'm also using the `PlayerSetting` to add the announces configured for Nuwani. I understand JS uses duck typing but it wouldn't be clear upon making changes in either classes.

I do wonder where the `player_setting_list` will belong. I don't feel like this can be seen as an entity?

I'm not sure if I understand you right on the last point however I do want to keep the `PlayerSettingsDatabase` to be responsible for the queries. I will make `PlayerSettingsCommands` responsible for changing the values and setting them upon log in.